sktime/datatypes/_check.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+"""Machine type checkers for scitypes.
+
+Exports
+-------
+check_is(obj, mtype: str, scitype: str)
+    checks whether obj is mtype for scitype
+    returns boolean yes/no and metadata
+
+check_raise(obj, mtype: str, scitype:str)
+    checks whether obj is mtype for scitype
+    returns True if passes, otherwise raises error
+
+mtype(obj, as_scitype: str = None)
+    infer the mtype of obj, considering it as as_scitype
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = [
+    "check_is",
+    "check_raise",
+    "mtype",
+]
+
+import numpy as np
+
+from typing import Union
+from sktime.datatypes._panel import check_dict_Panel
+from sktime.datatypes._series import check_dict_Series
+from sktime.datatypes._registry import mtype_to_scitype
+
+# pool convert_dict-s
+check_dict = dict()
+check_dict.update(check_dict_Series)
+check_dict.update(check_dict_Panel)
+
+
+def check_is(
+    obj,
+    mtype: str,
+    scitype: Union[str, list] = None,
```

Comment:
```suggestion
    mtype: Union[str, list[str]],
    scitype: str = None,
```
* I think type hints were mixed up between mtype and scitype
* I think you should be specific that it is a list of string, not just any list